### PR TITLE
feat: add Form Header

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Description
Introduce Form Header which is needed for Wizard Summary page

Note:
--sapFontHeaderWeight is not present in sap-theming, the designers have been contacted about this.

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [N/A] hover state of the element follow design spec
- [N/A] focus state of the element follow design spec
- [N/A] active state of the element follow design spec
- [N/A] selected state of the element follow design spec
- [N/A] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [N/A] Responsiveness rules - the component has modifier classes for all breakpoints
- [N/A] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [N/A] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by salarenko — APPROVED** [review]
Looks good. I'm just wondering if we shouldn't have an example how to use form with form-header.

**Comment by InnaAtanasova** [comment]
> Looks good. I'm just wondering if we shouldn't have an example how to use form with form-header.

There will be one in the Wizard Summary page

File: src/form-header.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+.#{$block} {
+  @include fd-reset();
+  @include fd-flex-vertical-center();
+
+  width: 100%;
+  height: 2.75rem;
+  background-color: var(--sapGroup_TitleBackground);
+  padding-left: $fd-form-header-padding-x;
+  padding-right: $fd-form-header-padding-x;
+  box-shadow: 0 var(--sapList_BorderWidth) var(--sapList_BorderWidth) var(--sapGroup_TitleBorderColor);
```
Comment (JKMarkowski) [inline src/form-header.scss:20]: image
Can you add new parameter to theming, so for high contrast there will be `3 * --sapList_BorderWidth` and for rest `--sapList_BorderWidth`?

**Review by JKMarkowski — COMMENTED** [review]
Great Job! I left one comment regarding theming, rest LGTM

**Review by jbadan — APPROVED** [review]
What a hilariously specific component 🤣

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Review by stefanoScalzo — APPROVED** [review]

**Pull request merged**